support-metrics-client/src/main/java/io/confluent/support/metrics/MetricsReporter.java

```diff
@@ -39,212 +34,80 @@
-   * it doesn't exist yet.
-   */
-  private static final int SUPPORT_TOPIC_PARTITIONS = 1;
-
-  /**
-   * Length of the wait period we give the server to start up completely (in a different thread)
-   * before we begin metrics collection.
-   */
-  private static final long SETTLING_TIME_MS = 10 * 1000L;
 
-  private final String customerId;
-  private final long reportIntervalMs;
-  private final String supportTopic;
-  private KafkaSubmitter kafkaSubmitter;
-  private final ConfluentSubmitter confluentSubmitter;
-  private final Collector metricsCollector;
-  private final AvroSerializer encoder = new AvroSerializer();
   private final KafkaServer server;
-  private final KafkaUtilities kafkaUtilities;
+  private final Runtime serverRuntime;
+  private final KafkaSupportConfig kafkaSupportConfig;
+  private final KafkaServerZkUtilsProvider zkUtilsProvider;
 
   public MetricsReporter(KafkaServer server,
-                         Properties serverConfiguration,
+                         KafkaSupportConfig kafkaSupportConfig,
                          Runtime serverRuntime) {
-    this(server, serverConfiguration, serverRuntime, new KafkaUtilities());
+
+    this(server, kafkaSupportConfig, serverRuntime, new KafkaUtilities());
   }
 
   /**
    * @param server              The Kafka server.
-   * @param serverConfiguration The properties this server was created from, plus extra Proactive Support (PS) ones
+   * @param kafkaSupportConfig  The properties this server was created from, plus extra Proactive
+   *                            Support (PS) ones
    *                            Note that Kafka does not understand PS properties,
    *                            hence server->KafkaConfig() does not contain any of them, necessitating
    *                            passing this extra argument to the API.
    * @param serverRuntime       The Java runtime of the server that is being monitored.
    * @param kafkaUtilities      An instance of {@link KafkaUtilities} that will be used to perform
    *                            e.g. Kafka topic management if needed.
    */
-  public MetricsReporter(KafkaServer server,
-                         Properties serverConfiguration,
-                         Runtime serverRuntime,
-                         KafkaUtilities kafkaUtilities) {
-    this.kafkaUtilities = kafkaUtilities;
-
-    if (server == null || serverConfiguration == null || serverRuntime == null || kafkaUtilities == null) {
+  public MetricsReporter(
+      KafkaServer server,
+      KafkaSupportConfig kafkaSupportConfig,
+      Runtime serverRuntime,
+      KafkaUtilities kafkaUtilities
+  ) {
+    super(kafkaSupportConfig, kafkaUtilities);
+    this.server = server;
+    this.serverRuntime = serverRuntime;
+    this.kafkaSupportConfig = kafkaSupportConfig;
+    this.zkUtilsProvider = new KafkaServerZkUtilsProvider(server);
+    if (server == null || kafkaSupportConfig == null || serverRuntime == null || kafkaUtilities == null) {
```

Comment:
It might be better here to check these individually, i.e,  `Objects.requireNonNull(server, "server cannot be null")` that way we know what was null